pkg/test/utils/kubernetes.go

```diff
@@ -584,36 +590,48 @@ func WithNamespace(obj runtime.Object, namespace string) runtime.Object {
 }
 
 // WithSpec applies the provided spec to the Kubernetes object.
-func WithSpec(obj runtime.Object, spec map[string]interface{}) runtime.Object {
-	return WithKeyValue(obj, "spec", spec)
+func WithSpec(t *testing.T, obj runtime.Object, spec map[string]interface{}) runtime.Object {
```

Comment:
I'm not very proud of this solution: passing in `t *testing.T` and failing the test here. But, the way this method and `WithStatus` and `WithLabels` below are used, makes it hard to do anything else meaningfully. At least we actually fail now if something goes wrong here.